ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/AuthHandlers.java

```diff
@@ -67,10 +67,16 @@ static void setupAuthHandlers(final Server server, final Router router,
               server.getConfig().getList(KsqlRestConfig.AUTHENTICATION_ROLES_CONFIG)),
           Provider.JAAS));
 
+      // pause before using the PLUGIN handlers because they can execute blocking code,
+      // which may cause the request to be read asynchronously
+      router.route().handler(AuthHandlers::pauseHandler);
```

Comment:
Per [this comment](https://github.com/confluentinc/ksql/pull/8975#discussion_r849919127), I think this extra pause call needs to be above the previous line (for the RoleBasedAuthZHandler) too?

Reply:
yeah let's just do that to be extra safe